### PR TITLE
There isn't keystore files UTC / JSON decoding

### PR DESCRIPTION
**Pull request by dulanov**
Impl. keystore files UTC / JSON decoding. As a part of #15.

### CODE REVIEW

File: src/address.rs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        let mut bytes = [0; ADDRESS_BYTES];
+
+        bytes.clone_from_slice(vec.as_slice());
+
+        Ok(Address(bytes))
+    }
+}
+
+impl ops::Deref for Address {
+    type Target = [u8; ADDRESS_BYTES];
```
Comment (dulanov) [inline src/address.rs]: here, probably better change to `[u8]`

File: src/address.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    ///
+    /// * `vec` - A byte vector with `ADDRESS_BYTES` length
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let addr = emerald::Address::try_from(vec![0; emerald::ADDRESS_BYTES]).unwrap();
+    /// assert_eq!(addr.to_string(), "0x0000000000000000000000000000000000000000");
+    /// ```
+    pub fn try_from(vec: Vec<u8>) -> Result<Self, AddressParserError> {
```
Comment (dulanov) [inline src/address.rs]: `Vec<u8>` > `&[u8]`

File: src/keystore/mod.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    pub dk_length: u32,
+    pub kdf: Kdf,
+    pub kdf_salt: [u8; KDF_SALT_BYTES],
+    pub keccak256_mac: [u8; KECCAK256_BYTES],
+    pub cipher_text: Vec<u8>,
+    pub cipher_iv: [u8; CIPHER_IV_BYTES],
+}
+
+/// Key derivation function
+#[derive(Clone, Debug, PartialEq, Eq)]
```
Comment (dulanov) [inline src/keystore/mod.rs]: + `Copy`?

File: tests/keystore/UTC--2015-08-11T06-13-53-359Z--008aeeda4d805471df9b2a5b0f38a0c3bcba786b [hunk 4]
```diff
@@ -1,20 +0,0 @@
...
-    cipher: 'aes-128-ctr',
-    ciphertext: '5318b4d5bcd28de64ee5559e671353e16f075ecae9f99c7a79a38af5f869aa46',
-    cipherparams: {
-      iv: '6087dab2f9fdbbfaddc31a909735c1e6'
-    },
-    mac: '517ead924a9d0dc3124507e3393d175ce3ff7c1e96529c6c555ce9e51205e9b2',
-    kdf: 'pbkdf2',
-    kdfparams: {
-      c: 262144,
-      dklen: 32,
```
Comment (splix) [inline tests/keystore/UTC--2015-08-11T06-13-53-359Z--008aeeda4d805471df9b2a5b0f38a0c3bcba786b]: This provides 256 bit key, but AES 128 uses only 128 bit of it

Comment (dulanov) [inline tests/keystore/UTC--2015-08-11T06-13-53-359Z--008aeeda4d805471df9b2a5b0f38a0c3bcba786b]: This key was generated by parity v1.5.11

**Pull request merged**